node/src/components/consensus/era_supervisor/tests.rs

```diff
@@ -62,66 +163,112 @@ fn new_test_chainspec(stakes: Vec<(PublicKey, u64)>) -> Chainspec {
-    let effects = handle(&mut es, event);
-    assert_eq!(effects.len(), 2); // Validate consensus value, request evidence against Alice.
-    let event = ClMessage::FinalizeBlock.received(NodeId(1), EraId(0));
-    let effects = handle(&mut es, event);
-    assert_eq!(effects.len(), 2); // Announce and execute block.
-    assert_eq!(&[alice_pk], &*es.active_eras[&EraId(0)].accusations());
+    let mut effects = handle(&mut es, event);
 
-    // TODO: Add a reactor to verify the effects.
+    // As a result, the era supervisor should request validation of the proto block and evidence
+    // against Alice.
+    let validate = tokio::spawn(effects.pop().unwrap());
+    let request_evidence = tokio::spawn(effects.pop().unwrap());
+    assert!(effects.is_empty());
+    reactor.expect_send_message(NodeId(1)).await; // Sending request for evidence.
+    request_evidence.await.unwrap();
+
+    // The block validator returns true: The deploys in the block are valid. As a result, we don't
+    // do anything yet (no new effects): We have to wait for the evidence first.
+    reactor
+        .expect_block_validation(&proto_block, NodeId(1), timestamp, true)
+        .await;
+    let rv_event = validate.await.unwrap().pop().unwrap();
+    let effects = handle(&mut es, rv_event);
+    assert!(effects.is_empty());
+
+    // Node 1 replies with evidence of Alice's fault. That completes our requirements for the
+    // proposed block. The era supervisor announces that it has passed the proposed block to the
+    // consensus protocol. It also gossips the new evidence to other nodes.
+    let event = ClMessage::Evidence(alice_pk).received(NodeId(1), EraId(0));
+    let mut effects = handle(&mut es, event);
+    let _announce_proposed = tokio::spawn(effects.pop().unwrap());
+    let broadcast_evidence = tokio::spawn(effects.pop().unwrap());
+    assert!(effects.is_empty());
+    assert_eq!(proto_block, reactor.expect_proposed().await);
+    // TODO: Why does it hang if we await this effect?
+    // announce_proposed.await.unwrap();
+    reactor.expect_broadcast().await; //Gossip evidence to other nodes.
+    broadcast_evidence.await.unwrap();
+
+    // Node 1 now sends us another message that is sufficient for the protocol to finalize the
+    // block. The era supervisor is expected to announce finalization, and to request execution.
+    let event = ClMessage::FinalizeBlock.received(NodeId(1), EraId(0));
+    let mut effects = handle(&mut es, event);
+    tokio::spawn(effects.pop().unwrap()).await.unwrap();
+    tokio::spawn(effects.pop().unwrap()).await.unwrap();
+    assert!(effects.is_empty());
+    let expected_fb = FinalizedBlock::new(
+        proto_block,
+        timestamp,
+        None, // not the era's last block
+        EraId(0),
+        0, // height
+        bob_pk,
+    );
+    assert_eq!(expected_fb, reactor.expect_execute().await);
+    assert_eq!(expected_fb, reactor.expect_finalized().await);
 
-    assert_eq!(EraId(0), es.current_era);
```

Comment:
It's still true that the current era id is 0, right?

Reply:
Yes, but I now check that by comparing the reported finalized block with `expected_finalized_block`, which also contains the era ID.